src/storage/src/metadata.rs

```diff
@@ -635,6 +644,7 @@ impl TryFrom<RegionDescriptor> for RegionMetadataBuilder {
             .name(desc.name)
             .id(desc.id)
             .row_key(desc.row_key)?
+            .compaction_time_window(None) // TODO should I update `RegionDescriptor` as well?
```

Comment:
@v0y4g3r Could you please give me a hint regarding this line?

Reply:
> @v0y4g3r Could you please give me a hint regarding this line?

`RegionDescriptor` and `RegionMetadata(Builder)` should be mutually convertible, thus I think you may have to add a `compaction_time_window: Option<i64>` field to `RegionDescriptor`:

https://github.com/GrepTimeTeam/greptimedb/blob/8ffc078f88f5e265eb0821820593a1c4bd309c17/src/store-api/src/storage/descriptors.rs#L140-L152